src/main/scala/org/broadinstitute/dsde/workbench/sam/service/PolicyEvaluatorService.scala

```diff
@@ -123,52 +124,62 @@ class PolicyEvaluatorService(
+      results <- if (rt.isAuthDomainConstrainable) {
+        populateAuthDomains(resourceTypeName, userId, userResourcesResponses, samRequestContext)
+      } else {
+        IO.pure(userResourcesResponses.map(_.some))
       }
     } yield results.flatten
 
+  private def populateAuthDomains(resourceTypeName: ResourceTypeName, userId: WorkbenchUserId, userResourcesResponses: Iterable[UserResourcesResponse], samRequestContext: SamRequestContext) = {
+    for {
+      resources <- accessPolicyDAO.listResourcesWithAuthdomains(resourceTypeName, userResourcesResponses.map(_.resourceId).toSet, samRequestContext)
+      userManagedGroups <- listUserManagedGroups(userId, samRequestContext)
+    } yield {
+      val authDomainMap = resources.map(resource => resource.resourceId -> resource.authDomain).toMap
+      userResourcesResponses.map { response =>
+        authDomainMap.get(response.resourceId) match {
+          case Some(resourceAuthDomains) =>
+            val userNotMemberOf = resourceAuthDomains -- userManagedGroups
+            response.copy(authDomainGroups = resourceAuthDomains, missingAuthDomainGroups = userNotMemberOf).some
+          case None =>
+            logger.error(s"Corrupted data. ${response.resourceId} should have auth domains defined")
+            none[UserResourcesResponse]
+        }
+      }
+    }
+  }
+
   private def getResourceType(resourceTypeName: ResourceTypeName): IO[ResourceType] = {
     IO.fromEither(
       resourceTypes
         .get(resourceTypeName)
         .toRight(new WorkbenchException(s"missing configration for resourceType ${resourceTypeName}")))
   }
 
-  def listUserManagedGroupsWithRole(userId: WorkbenchUserId, samRequestContext: SamRequestContext): IO[Set[ManagedGroupAndRole]] =
-    for {
-      ripns <- accessPolicyDAO.listAccessPolicies(ManagedGroupService.managedGroupTypeName, userId, samRequestContext)
-    } yield {
-      ripns
-        .filter(ripn => ManagedGroupService.userMembershipPolicyNames.contains(ripn.accessPolicyName))
-        .map(p => ManagedGroupAndRole(WorkbenchGroupName(p.resourceId.value), ManagedGroupService.getRoleName(p.accessPolicyName.value)))
+  def listUserManagedGroupsWithRole(userId: WorkbenchUserId, samRequestContext: SamRequestContext): IO[Iterable[ManagedGroupAndRole]] =
+    accessPolicyDAO.listUserResourcesWithRolesAndActions(ManagedGroupService.managedGroupTypeName, userId, samRequestContext) map { accessibleGroupResources =>
+      for {
+        groupResource <- accessibleGroupResources
+        roleName <- groupResource.allRolesAndActions.roles.collect {
+          case MangedGroupRoleName(roleName) if ManagedGroupService.userMembershipRoleNames.contains(roleName) => roleName
+        }
+      } yield {
+        ManagedGroupAndRole(WorkbenchGroupName(groupResource.resourceId.value), roleName)
+      }
```

Comment:
reimplemented to use `listUserResourcesWithRolesAndActions` instead of `listAccessPolicies`